Add a SLIP client for OSC over serial ports.

MartijnBraam opened the pull request:
The OSC specification lets you send messages over any framed data channel and using the framing part of SLIP seems to be the most popular solution for devices allowing OSC control over USB.

attwad commented on pythonosc/slip_client.py:
```diff
@@ -0,0 +1,47 @@
+        encoded += self.END
+        self.ser.write(encoded)
+
+    def receive(self):
```
can you please add some docs on what to expect as return type please (OscMessage)

attwad commented on pythonosc/slip_client.py:
```diff
@@ -0,0 +1,47 @@
+        while True:
+            c = self.ser.read(1)
+            if c is None:
+                return None
```
just `return` please, no need for None

attwad commented on pythonosc/slip_client.py:
```diff
@@ -0,0 +1,47 @@
+            if c == self.ESC:
+                c = self.ser.read(1)
+                if c is None:
+                    return None
```
same here

attwad reviewed (commented):
Seems like tests are unhappy with the serial import, I don't think this is part of the standard library right?  You could add the necessary dependency to https://github.com/attwad/python-osc/blob/master/.github/workflows/python-test.yml to make them pass.

MartijnBraam commented:
Fixed up the receive method and added pyserial as dependency.

I wonder if it's better to return an exception instead of None when the serial port dissapears. The upside of None is that it follows the socket API a bit more closely.

attwad commented:
> Fixed up the receive method and added pyserial as dependency.
> 
> I wonder if it's better to return an exception instead of None when the serial port dissapears. The upside of None is that it follows the socket API a bit more closely.

Fine as is imo, thanks.

Tests are still failing but I don't think you want to write a whole set of stub types for the pyserial package... fine to suppress with # type: ignore imo. Cf. https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports

MartijnBraam commented:
Seems like there are also errors in files I haven't touched and some tests complaining about the removed None (on 3.7)
